src/middlewares/Fetch.spec.ts

```diff
@@ -2,20 +2,27 @@ import {IRequest, IRequestHeaders} from "../Client";
 import {Fetch} from "./Fetch";
 
 describe("Fetch", () => {
+    describe("is defined and is available", () => {
+        it("should be defined", () => {
+            const fetch = new Fetch();
+            expect(fetch).toBeDefined();
+        });
+    });
     describe("process()", () => {
         it("is defined", () => {
             const obj = new Fetch();
             expect(obj.process).toBeDefined();
         });
 
-        it("throws on non-object options", () => {
-            const obj = new Fetch();
-            expect(() => obj.process(false as any, null as any)).toThrow();
-        });
-
-        it("throws on null options", () => {
-            const obj = new Fetch();
-            expect(() => obj.process(null as any, null as any)).toThrow();
+        describe("Should throw on invalid options", () => {
```

Comment:
Again: Think of reading it as a sentence... this one now reads:

_Fetch process() Should throw on invalid options throws on non-object options._

You could either
- unwrap the ShouldThrow-describe

or if you intend to have many throw-tests and want to group them
- just call it describe it as "throws", "should throws" or something similarly short.

First option is ok if there's only a handful of cases anyway, last one if there are many, though this might be a good indicator of a function being too complex and therefor splitworthy anyway. Unwrapping would read as something like "Fetch" "process()" "throws on non-object options" which is nice and easy :) 

There might be a perfectly good reason though for a complex function with many throw test cases in which applying the second option and therefor having a structure like "Fetch process() throws"... "without parameters", "on omitted options", "on missmatching id", etc. etc. will help readability without cluttering console :)